dev/NumberBox/InteractionTests/NumberBoxTests.cs

```diff
@@ -245,13 +245,94 @@ public void CoersionTest()
             }
         }
 
+        [TestMethod]
+        public void BasicCalculationTest()
+        {
+            using (var setup = new TestSetupHelper("NumberBox Tests"))
+            {
+                RangeValueSpinner numBox = FindElement.ByName<RangeValueSpinner>("TestNumberBox");
+
+                Log.Comment("Verify that calculations don't work if AcceptsCalculations is false");
+                EnterText(numBox, "5 + 3");
+                Verify.AreEqual(0, numBox.Value);
+
+                Check("CalculationCheckBox");
+
+                int numErrors = 0;
+                const double resetValue = 1234;
+                
+                Dictionary<string, double> expressions = new Dictionary<string, double>
+                {
+                    // Valid expressions. None of these should evaluate to the reset value.
+                    { "5", 5 },
+                    { "-358", -358 },
+                    { "12.34", 12.34 },
+                    { "5 + 3", 8 },
+                    { "12345 + 67 + 890", 13302 },
+                    { "000 + 0011", 11 },
+                    { "5 - 3 + 2", 4 },
+                    { "3 + 2 - 5", 0 },
+                    { "9 - 2 * 6 / 4", 6 },
+                    { "9 - -7",  16 },
+                    { "9-3*2", 3 },         // no spaces
+                    { " 10  *   6  ", 60 }, // extra spaces
+                    { "10 /( 2 + 3 )", 2 },
+                    { "5 * -40", -200 },
+                    { "(1 - 4) / (2 + 1)", -1 },
+                    { "3 * ((4 + 8) / 2)", 18 },
+                    { "23 * ((0 - 48) / 8)", -138 },
+                    { "((74-71)*2)^3", 216 },
+                    { "2 - 2 ^ 3", -6 },
+                    { "2 ^ 2 ^ 2 / 2 + 9", 17 },
+                    { "5 ^ -2", 0.04 },
+                    { "5.09 + 14.333", 19.423 },
+                    { "2.5 * 0.35", 0.875 },
+                    { "-2 - 5", -7 },       // begins with negative number
+                    { "(10)", 10 },         // number in parens
+                    { "(-9)", -9 },         // negative number in parens
+
+                    // These should not parse, which means they will reset back to the previous value.
+                    { "5x + 3y", resetValue },        // invalid chars
+                    { "5 + (3", resetValue },         // mismatched parens
+                    { "9 + (2 + 3))", resetValue },
+                    { "(2 + 3)(1 + 5)", resetValue }, // missing operator
+                    { "9 + + 7", resetValue },        // extra operators
+                    { "9 - * 7",  resetValue },
+                    { "9 - - 7",  resetValue },
```

Comment:
>9 - - 7 [](start = 23, length = 7)

Should this be an error?

Reply:
In terms of the way the DecimalParser handles it, "-7" is -7 and "- 7" is not a number. So I have the positive test "9 - -7" evaluate to 16, but "9 - - 7" is not a valid expression.

Reply:
seems like a good solution

---
In reply to: [345453668](https://github.com/microsoft/microsoft-ui-xaml/pull/1602#discussion_r345453668) [](ancestors = 345453668)